README.md
emailpoubelle
=============
A Yunohost version of Email Poubelle by David Mercereau. All thanks to David. 
http://www.mercereau.info/sortie-de-la-version-1-0-demailpoubelle-php-email-jetable-auto-hebergeable/
=============

WARNING
=========
Using this will cause to disable the Yunohost ldap aliases ! Once installed, you won't be able to use the aliasses settings that you can see when you modify your personnal settings in the SSOWAT pannel.
Of course, once uninstalled, everything get back to normal. 

TODO : 
------
Test it ! (and check if that doesn't interfer with postfix and its aliases)

English traduction (at least)

cron job to remove redirections (0 */2 * * * /usr/bin/wget -q -t 1 -T 7200 -O /dev/null 'https://domain/poubelle/index.php?act=cron' >/dev/null 2>&1) conflict with SSOWAT + non-public app


src/CHANGELOG.md
## 1.1 (future)

  - support postfix-mysql (not plain text virtual file)
  - admin panel page with statistic
  - limit nb alias / email
  - javascript checkupgrade in admin page
  - add second table in database (for domain/checkupdate/intervale cron)
  - send email for advertisement action ?

## 1.0.1 (2014-01-06)

Bugfixes:

  - fixe ListeAlias no send email if error

## 1.0 (2013-10-02)

Features:

  - add database with PDO (remove plain text) 
		support mysql, sqlite, postgresql...
  - add multi-domain support
  - add memory email (with cookies)
  - add life for alias (optional)
  - add comment for alias (optional)
  - add template for example
  - add javascript in form (noscript compatible)
  - add cron for expir life email
  - add maintenance mode
  - add basic form anti-spam
  - add disable/enable alias function
  - pass UTF-8 encode

## 0.3 (2013-08-08) (without database)

  - add blacklist.txt regex
  - add function "alias list" 
  - add shell statistique script 
  - add readme

## 0.2 (2012-08-05)

Features:

  - add aliasdeny.txt regex
  - migrate to Net_DNS2
  - check email exist with DNS (check MX)

Bugfixes:

   - fixe http://forge.zici.fr/p/emailpoubelle-php/issues/4/

## 0.1b (2012-03-20)

  - start project



src/conf-dist.php
<?php

//-----------------------------------------------------------
// emailPoubelle config
// Licence : GNU GPL v3 : http://www.gnu.org/licenses/gpl.html
// Créateur : David Mercereau - david [aro] mercereau [.] info
// Home : http://poubelle.zici.fr
//----------------------------------------------------------- 

// writable for script
define('DATA', '../var');
// include directory
define('INC', '../lib');

// include function
include_once(INC.'/ep_function.php');

define('DEBUG', false);

// Domain email (separe with ; exemple : exemple.com;zici.fr;exemple.fr)
define('DOMAIN', 'exemple.fr');
//define('DOMAIN', 'exemple.com;zici.fr;exemple.fr');

// PDO stucture
// Exemple pour MYSQL : 
      define('DB', 'mysql:host=127.0.0.1;dbname=emailPoubelle');
      define('DBUSER', 'emailPoubelle');
      define('DBPASS', 'motdepassedefou');
// Exemple pour Sqlite : 
//      define('DB', 'sqlite:./data/emailPoubelle.sqlite');
//define('DB', 'sqlite:'.DATA.'/database.sdb');
// table prefix
define('DBTABLEPREFIX', 'ep_');

// Use cron (true/false) for life expire email (recomanded)
// 	If use true add in crontab : 
//	0 */2 * * * /usr/bin/wget -q -t 1 -T 7200 -O /dev/null 'http://exemple.com/page.php?act=cron' >/dev/null 2>&1
define('CRON', false);

// Fichier d'alias postfix
define('FICHIERALIAS', DATA.'/virtual');
define('BIN_POSTMAP', '/usr/sbin/postmap');

//test if we are in secure mode
$s = empty($_SERVER["HTTPS"]) ? '' : ($_SERVER["HTTPS"] == "on") ? "s" : "";
define('URLPAGE', 'http'.$s.'://'.$_SERVER["SERVER_NAME"].$_SERVER["SCRIPT_NAME"]);

// Email 
define('EMAILTAGSUJET', '[EmailPoubelle]');
// From de l'email
define('EMAILFROM', '"NO REPLY emailPoubelle" <emailpoubelle@exemple.fr>');
define('EMAILEND', 'exemple.fr');

// Alisas interdit : (regex ligne par ligne) - commenter pour désactiver
define('ALIASDENY', DATA.'/aliasdeny.txt');

// Blackliste d'email : (regex ligne par ligne) - commenter pour désactiver
define('BLACKLIST', DATA.'/blacklist.txt');

// Depend pear Net/DNS2
define('CHECKMX', false);
if (CHECKMX) {
    require_once('Net/DNS2.php');
    // Serveur DNS pour la résolution/vérification du nom de domaine
    define('NS1', 'ns1.fdn.org');
    define('NS2', '8.8.8.8');
}

// A indiquer si vous utiliser les URL's rewriting
// Exemple avec un htaccess
// 		RewriteRule ^ep-([a-z]+)-([a-zA-Z0-9+/=]+)\.html$  switch.php?page=emailPoubelle&act=$1&value=$2  [L]
//define('URLREWRITE_START', 'http://poubelle.zici.fr/ep-');
//define('URLREWRITE_MIDDLE', '-');
//define('URLREWRITE_END', '.html');
// Désactiver
define('URLREWRITE_START', false);
define('URLREWRITE_MIDDLE', false);
define('URLREWRITE_END', false);

// check update :
// 		enable : in seconds
//		disable : false
define('CHECKUPDATE', 86400);

// Maintenance mode 
define('MAINTENANCE_MODE', false);
// No maintenance for this ip (admin ip)
define('MAINTENANCE_IP', '10.0.0.1');

?>


src/www/index.php
<!DOCTYPE html>
<html lang="fr">
<head>
<title>emailPoubelle - Template-exemple</title>
<meta charset="UTF-8">
<meta name="description" content="">
<meta name="robots" content="index,follow,noarchive">
<link rel="stylesheet" href="template-exemple/style.css">
<link rel="shortcut icon" href="https://matlink.fr/poubelle/favicon.png"/>
<!--[if IE]><script src="template-exemple/html5-ie.js"></script><![endif]--> 

<meta name = "viewport" content = "user-scalable=no, initial-scale=1.0, maximum-scale=1.0, width=device-width">
<meta name="apple-mobile-web-app-capable" content="yes" />
<meta name="apple-mobile-web-app-status-bar-style" content="black" />
<link rel="apple-touch-icon" href="/m/images/iphone.png" />
<meta name="format-detection" content="telephone=no" />
<!--  A free Google web font embed because android does not have the browser safe fonts -->
<link rel="stylesheet" type="text/css" href="http://fonts.googleapis.com/css?family=Allerta">
<!-- Style sheet for NOT smart devices like older web phones  -->
<link href="/m/css/handheld.css" rel="stylesheet" type="text/css" media="handheld" />
<!-- Style sheet for smartphones and tablet devices -->
<link rel="stylesheet" href="/m/css/ios.css" media="screen" />
<!-- Style sheet for lousy incompatible frustrating IE mobile -->
<!--[if IEMobile]>
<link rel="stylesheet" href="/m/css/iemobile.css" media="screen" />
<![endif]-->
<script type="text/javascript">
window.scrollTo(0,1);
</script>







</head>
<body>
<!--[if lte IE 6]> 
<div class="alert-ie6" style="padding: 1em; background: #900; font-size: 1.1em; color: #fff;">
      	<p><strong>Attention ! </strong> Votre navigateur (Internet Explorer 6 ou 7) présente de sérieuses lacunes en terme de sécurité et de performances, dues à son obsolescence (il date de 2001).<br />En conséquence, ce site sera consultable mais de manière moins optimale qu'avec un navigateur récent (<a href="http://www.browserforthebetter.com/download.html" style="color: #fff;">Internet Explorer 8</a>, <a href="http://www.mozilla-europe.org/fr/firefox/" style="color: #fff;">Firefox 3</a>, <a href="http://www.google.com/chrome?hl=fr" style="color: #fff;">Chrome</a>, <a href="http://www.apple.com/fr/safari/download/" style="color: #fff;">Safari</a>,...)</p>
      </div>
<![endif]-->  
	<div id="a" style="float:left">
		<div id="b">
			<article style="float:left">
				<h1>Emails poubelle libre</h1>
				<p>Générer des emails poubelle sans contrainte de durée de vie. </p>
				<?php 
				// Intégration dans votre site :
				if (file_exists('../conf.php')) {
					include('../conf.php');
				} else {
					include('../conf-dist.php');
				}
				include('../emailPoubelle.php'); 
				?>
		</div>
</div>
</body>
</html>


src/emailPoubelle.php
<?php

//-----------------------------------------------------------
// Title : Email Poubelle
// Licence : GNU GPL v3 : http://www.gnu.org/licenses/gpl.html
// Author : David Mercereau - david [aro] mercereau [.] info
// Home : http://poubelle.zici.fr
// Date : 08/2013
// Version : 1.0
// Depend : Postifx (postmap command) php-pdo
//----------------------------------------------------------- 

// @todo
// 	form ergonomie
// 	sqlite
//	disable time	

//////////////////
// Init & check
//////////////////

define('VERSION', '1.0');

if (DEBUG) {
    error_reporting(E_ALL);
    ini_set('display_errors', 'On');
	echo '<div class="highlight-2">Debug activé <br />';
	echo print_r($_REQUEST);
	echo '</div>';
}

if (!defined('DOMAIN') || !defined('DATA') || !defined('DEBUG') || !defined('FICHIERALIAS') || !defined('DB')) {
	echo '<div class="highlight-1">Erreur : Il ne semble pas que le fichier de configuration conf.php soit inclue car les constantes ne sont pas présentes.</div>';
// check writable work directory
} else if (!is_writable(DATA)) {
	echo '<div class="highlight-1">Erreur : le répertoire de travail ne peut pas être écrit. Merci de contacter l\'administrateur</div>';
// check alias file is_writable 
} else if (!is_writable(FICHIERALIAS)) {
	echo '<div class="highlight-1">Erreur : le fichier d\'alias ne peut pas être écrit. Merci de contacter l\'administrateur</div>';
// check blacklist file is_writable
} else if (defined('BLACKLIST') && !is_readable(BLACKLIST)) {
    echo '<div class="highlight-1">Erreur : un fichier de blacklist est renseigné mais n\'est pas lisible. Merci de contacter l\'administrateur</div>';
// check aliasdeny file is_writable
} else if (defined('ALIASDENY') && !is_readable(ALIASDENY)) {
    echo '<div class="highlight-1">Erreur : un fichier d\'alias interdit est renseigné mais n\'est pas lisible. Merci de contacter l\'administrateur</div>';
// maintenance mod
} else if (MAINTENANCE_MODE == true && MAINTENANCE_IP != $_SERVER["REMOTE_ADDR"]) {
	echo '<div class="highlight-2">Le service est en maintenance.</div>';
} else {

if (MAINTENANCE_MODE == true) {
	echo '<div class="highlight-2">Le service est en maintenance.</div>';
}

// Connect DB
try {
	if (preg_match('/^sqlite/', DB)) {
		$dbco = new PDO(DB);
	} else {
		$dbco = new PDO(DB, DBUSER, DBPASS);
	}
	$dbco->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
} catch ( PDOException $e ) {
	die('Connexion à la base '.$e->getMessage());
}
// Create DB if not exists
try {
	// status : 0=not verified - 3=disable - 5=active
	if (preg_match('/^sqlite/', DB)) {
		$create = $dbco->query("CREATE TABLE IF NOT EXISTS ".DBTABLEPREFIX."alias (
								id INTEGER PRIMARY KEY,
								status INTEGER(1) NOT NULL,
								alias CHAR(150) NOT NULL UNIQUE,
								email CHAR(150) NOT NULL,
								dateCreat DATETIME NOT NULL,
								dateExpir DATETIME,
								comment TEXT);");
	} else {
		$create = $dbco->query("CREATE TABLE IF NOT EXISTS ".DBTABLEPREFIX."alias (
								id INTEGER PRIMARY KEY  AUTO_INCREMENT,
								status INTEGER(1) NOT NULL,
								alias CHAR(150) NOT NULL UNIQUE,
								email CHAR(150) NOT NULL,
								dateCreat DATETIME NOT NULL,
								dateExpir DATETIME,
								comment TEXT);");
	}
} catch ( PDOException $e ) {
	echo '<div class="highlight-1">Erreur à l\'initialisation des tables. Merci de contacter l\'administrateur ';
	if (DEBUG) { $e->getMessage(); }
	echo '</div>';
	die();
}

//////////////////
// Start program
//////////////////

// get process "act" (action)
$action = isset($_GET['act']) ? $_GET['act'] : '';
switch ($action) {
	case "validemail" :
		$get_value = urlUnGen($_GET['value']);
		if ($dbco->query("SELECT COUNT(*) FROM ".DBTABLEPREFIX."alias WHERE id = '".$get_value['id']."' AND status = 0")->fetchColumn() != 0) {
			UpdateStatusAlias($get_value['id'], $get_value['alias_full'], 5);
			echo '<div class="highlight-3">Votre email poubelle <b>'.$get_value['alias_full'].'</b> est maintenant actif</div>';
		} else {
			echo '<div class="highlight-1">Erreur : ID introuvable ou déjà validé</div>';
		}
	break;
	case "disable" :
		$get_value = urlUnGen($_GET['value']);
		DisableAlias($get_value['id'], $get_value['alias_full'], null);
	break;
	case "enable" :
		$get_value = urlUnGen($_GET['value']);
		EnableAlias($get_value['id'], $get_value['alias_full'], null);
	break;
	case "delete" :
		$get_value = urlUnGen($_GET['value']);
		DeleteAlias($get_value['id'], $get_value['alias_full']);
	break;
	case "cron" :
		if (CRON) {
			echo '<div class="highlight-2">La tâche planifié est lancé</div>';
			LifeExpire();
		} else {
			echo '<div class="highlight-1">Vous n\'avez pas autorisé le lancement par tâche planifié</div>';
		}
	break;
}
// Form
if (isset($_POST['username']) && $_POST['username'] != '') { // minimal anti-spam 
	echo 'Hello you';
} else if (isset($_POST['list'])) {
	$email=strtolower(StripCleanToHtml($_POST['email']));
	if (! filter_var($email, FILTER_VALIDATE_EMAIL)) {
		echo '<div class="highlight-1">Erreur : Adresse email incorrect</div>';
	} else if (! VerifMXemail($email)) {
		echo '<div class="highlight-1">Erreur : Adresse email incorrect (2)</div>';
	} else if (ListeAlias($email)) {
		echo '<div class="highlight-3">Un email vient de vous être envoyé</div>';
	} else {
		echo '<div class="highlight-1">Erreur : aucun email actif connu</div>';
	}
} else if (isset($_POST['email']) && isset($_POST['alias'])) {
	$alias=strtolower(StripCleanToHtml($_POST['alias']));
	$email=strtolower(StripCleanToHtml($_POST['email']));
	$domain=StripCleanToHtml($_POST['domain']);
	$life=$_POST['life'];
	$comment=StripCleanToHtml($_POST['comment']);
	$alias_full=$alias.'@'.$domain;
	// Check form
	if (! filter_var($email, FILTER_VALIDATE_EMAIL)) {
		echo '<div class="highlight-1">Erreur : Adresse email incorrect</div>';
	} else if (! VerifMXemail($email)) {
		echo '<div class="highlight-1">Erreur : Adresse email incorrect (2)</div>';
	} else if (! preg_match('#^[\w.-]+$#',$alias)) {
		echo '<div class="highlight-1">Erreur : Format de l\'email poubelle incorrect</div>';
	} else if (! preg_match('#'.$domain.'#',DOMAIN)) {
		echo '<div class="highlight-1">Erreur : ce domain n\'est pas pris en charge</div>';
	} else if (AliasDeny($alias)) {
		echo '<div class="highlight-1">Erreur : email poubelle interdit</div>';
	} else if (BlacklistEmail($email)) {
		echo '<div class="highlight-1">Erreur : vous avez été blacklisté sur ce service</div>';
	// add 
	} elseif (isset($_POST['add'])) {
		if ($dbco->query("SELECT COUNT(*) FROM ".DBTABLEPREFIX."alias WHERE alias = '".$alias_full."'")->fetchColumn() != 0) {
			echo '<div class="highlight-1">Erreur : cet email poubelle est déjà utilisé</div>';
		} else {
			if ($dbco->query("SELECT COUNT(*) FROM ".DBTABLEPREFIX."alias WHERE email = '".$email."' AND status > 0")->fetchColumn() != 0) {
				AjouterAlias(5, $alias_full, $email, $life, $comment);
				echo '<div class="highlight-3">Votre email poubelle <b>'.$alias_full.' > '.$email.'</b> est maintenant actif</div>';
			} else {
				$lastId=AjouterAlias(0, $alias_full, $email, $life, $comment);
				$message= "Confirmation de la création de votre redirection email poubelle : ";
				$message= $alias_full.' => '.$email."\n";
				$message= "Cliquer sur le lien ci-dessous pour confirmer : \n";
				$message.= "\t * ".urlGen('validemail',$lastId,$alias_full)."\n";
				$message.= "\n";
				$message.= "Pour supprimer cet email poubelle vous pouvez vous rendre sur le lien ci-dessou : \n";
				$message.= "\t * ".urlGen('delete',$lastId,$alias_full)."\n";
				$message.= "\n";
				$message.= "Après confirmation, vous pourez suspendre temporairement cet email poubelle vous pouvez vous rendre sur le lien ci-dessou : \n";
				$message.= "\t * ".urlGen('disable',$lastId,$alias_full)."\n";
				SendEmail($email,'Confirmation alias '.$alias,$message);
				echo '<div class="highlight-2">Votre email ('.$email.') nous étant inconnu, une confirmation vous a été envoyé par email.</div>';
			}
		}
	// delete
	} else if (isset($_POST['del'])) {
		if ($id = $dbco->query("SELECT id FROM ".DBTABLEPREFIX."alias WHERE email = '".$email."' AND alias = '".$alias_full."'")->fetchColumn()) {
			$message= "Confirmation de la création de votre redirection email poubelle : ";
			$message= $alias_full.' => '.$email."\n";
			$message= "Cliquer sur le lien ci-dessous pour confirmer la suppression : \n";
			$message.= "\t * ".urlGen('delete',$id,$alias_full)."\n\n";
			$message.= "Sinon pour suspendre temporairement cet email poubelle vous pouvez vous rendre sur le lien ci-dessou : \n";
			$message.= "\t * ".urlGen('disable',$id,$alias_full)."\n";
			SendEmail($email,'Suppression de l\'alias '.$alias,$message);
			echo '<div class="highlight-2">Un email de confirmation vient de vous être envoyé.</div>';
		} else {
			echo '<div class="highlight-1">Erreur : impossible de trouver cet email poubelle</div>';
		}
	// disable
	} else if (isset($_POST['disable'])) {
		DisableAlias(null, $alias_full, $email);
	// enable
	} else if (isset($_POST['enable'])) {
		EnableAlias(null, $alias_full, $email);
	}

	// memory email
	if (isset($_POST['memory'])) {
		setcookie ("email", StripCleanToHtml($email), time() + 31536000);
	} else if (isset($_COOKIE['email'])) {
		unset($_COOKIE['email']);
	}
}

//////////////////
// Printing form
//////////////////

?>

<form action="<?= URLPAGE?>" method="post">
<div id="onglet" style="display: none;">
	<input type="button" value="Ajouter" id="onglet-add" onClick="ongletChange(this.id)" /> 
	<input type="button" id="onglet-list" value="Lister" onClick="ongletChange(this.id)" /> 
	<input type="button" id="onglet-del" value="Supprimer" onClick="ongletChange(this.id)" /> 
	<input type="button" id="onglet-dis" value="Suspendre" onClick="ongletChange(this.id)" />
	<input type="button" id="onglet-en" value="Reprendre" onClick="ongletChange(this.id)" />
	<input type="hidden" name="onglet-actif" id="onglet-actif" value="onglet-add" />
</div>
<div id="form-email">
	<label for="email">Votre email réel : </label>
	<input type="text" name="email" <?php if (isset($_COOKIE['email'])) { echo 'value="'.$_COOKIE['email'].'"'; } ?> id="input-email" size="24" border="0"  onkeyup="printForm()" onchange="printForm()"  /> 
	<input class="button2" type="submit" name="list" id="button-list" value="Lister" />
	<input type="checkbox" name="memory" id="check-memory" <?php if (isset($_COOKIE['email'])) { echo 'checked="checked" '; } ?>/> Mémoriser
</div>
<div id="form-alias">
	<label for="alias">Nom de l'email poubelle : </label>
	<input type="text" name="alias" id="input-alias" size="24" border="0" onkeyup="printForm()" onchange="printForm()" placeholder="Ex : jean-petiteannonce" /> @<?php
		$domains = explode(';', DOMAIN);
		if (count($domains) == 1) {
			echo DOMAIN.'<input type="hidden" value="'.DOMAIN.'" name="domain" id="input-domain" />';
		} else {
			echo '<select name="domain" id="input-domain">';
			foreach ($domains as $one_domain)  {
				echo '<option value="'.$one_domain.'">'.$one_domain.'</option>';
			}
			echo '</select>';
		}
	?>
	<select name="life" id="input-life">
		<option value="0">Illimité</option>
		<option value="7200">2 heure</option>
		<option value="21600">6 heures</option>
		<option value="86400">1 jour</option>
		<option value="604800">7 jours</option>
		<option value="1296000">15 jours</option>
		<option value="2592000">30 jours</option>
		<option value="7776000">90 jours</option>
	</select>
</div>
<div id="form-comment">
	<label for="comment">Un commentaire pour l'ajout ? (pour votre mémoire)</label>
	<input type="text" name="comment" size="54" placeholder="Ex : Inscription sur zici.fr" />
</div>
<div id="form-submit">
	<input class="button" type="submit" id="button-add" name="add" value="Activer" />
	<input class="button" type="submit" id="button-del" name="del" value="Supprimer" />
	<input class="button" type="submit" id="button-enable" name="enable" value="Reprendre" />
	<input class="button" type="submit" id="button-disable" name="disable" value="Susprendre" />
</div>
<div id="lePecheur" style="display: none;">
	<input name="username" type="text" />
</div>
</form>

<script type="text/javascript">
	function validateEmail(email) { 
		var re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
		return re.test(email);
	} 
	function printForm() {
		if (validateEmail(document.getElementById('input-email').value) && document.getElementById('input-alias').value != '') {
			document.getElementById('input-alias').disabled = false; 
			document.getElementById('input-domain').disabled = false; 
			document.getElementById('button-list').disabled = false; 
			document.getElementById('button-add').disabled = false; 
			document.getElementById('button-disable').disabled = false; 
			document.getElementById('button-enable').disabled = false; 
			document.getElementById('button-del').disabled = false; 
			document.getElementById('input-life').disabled = false; 
			if (document.getElementById('onglet-actif').value == 'onglet-add') {
				document.getElementById('form-comment').style.display = "block" ;
			}
		} else if (validateEmail(document.getElementById('input-email').value)) {
			document.getElementById('input-alias').disabled = false; 
			document.getElementById('input-domain').disabled = false; 
			document.getElementById('button-list').disabled = false;
			document.getElementById('input-life').disabled = false;
			document.getElementById('form-comment').style.display = "display" ;
			document.getElementById('button-add').disabled = true; 
			document.getElementById('button-disable').disabled = true; 
			document.getElementById('button-enable').disabled = true; 
			document.getElementById('button-del').disabled = true; 
			document.getElementById('input-life').disabled = true;
			document.getElementById('form-comment').style.display = "none" ;
		} else {
			document.getElementById('input-alias').disabled = true; 
			document.getElementById('input-domain').disabled = true; 
			document.getElementById('button-list').disabled = true; 
			document.getElementById('button-add').disabled = true; 
			document.getElementById('button-disable').disabled = true; 
			document.getElementById('button-enable').disabled = true; 
			document.getElementById('button-del').disabled = true; 
			document.getElementById('input-life').disabled = true;
			document.getElementById('form-comment').style.display = "none" ;
		}
	}
	function ongletPrint() {
		var ongletActif = document.getElementById('onglet-actif').value;
		document.getElementById('onglet-add').className = "close" ;
		document.getElementById('onglet-del').className = "close" ;
		document.getElementById('onglet-list').className = "close" ;
		document.getElementById('onglet-en').className = "close" ;
		document.getElementById('onglet-dis').className = "close" ;
		document.getElementById(ongletActif).className = "open" ;
		document.getElementById('input-life').style.display = "none" ;
		document.getElementById('form-alias').style.display = "inline-block" ;
		document.getElementById('button-add').style.display = "none" ;
		document.getElementById('button-del').style.display = "none" ;
		document.getElementById('button-list').style.display = "none" ;
		document.getElementById('button-disable').style.display = "none" ;
		document.getElementById('button-enable').style.display = "none" ;
		if (ongletActif == 'onglet-add') {
			document.getElementById('button-add').style.display = "inline-block" ;
			document.getElementById('input-life').style.display = "inline-block" ;
		} else if (ongletActif == 'onglet-del') {
			document.getElementById('button-del').style.display = "inline-block" ;
		} else if (ongletActif == 'onglet-en') {
			document.getElementById('button-enable').style.display = "inline-block" ;
		} else if (ongletActif == 'onglet-dis') {
			document.getElementById('button-disable').style.display = "inline-block" ;
		} else if (ongletActif == 'onglet-list') {
			document.getElementById('button-list').style.display = "inline-block" ;
			document.getElementById('form-alias').style.display = "none" ;
		}
	}
	function ongletChange(ongletValue) {
		document.getElementById('onglet-actif').value = ongletValue;
		ongletPrint();
	}
	document.getElementById('onglet').style.display = "block" ;
	ongletPrint();
	printForm();
</script>
<p>Version <?= VERSION ?> - Créé par David Mercereau sous licence GNU GPL v3</p>
<p>Télécharger et utiliser ce script sur le site du projet <a target="_blank" href="http://forge.zici.fr/p/emailpoubelle-php/">emailPoubelle.php</a></p>

<?php 
// execute lifeExpir if isn't in crontab
if (!CRON) { LifeExpire(); }
// Close connexion DB
$dbco = null;
// checkupdate
echo CheckUpdate(); 
} // end maintenance mod
?>
